code/framework/java-server/src/main/java/io/github/ibuildthecloud/gdapi/request/parser/DefaultApiRequestParser.java

```diff
@@ -160,22 +162,76 @@ protected String parseClientIp(ApiRequest apiRequest, HttpServletRequest request
         String clientIp = request.getRemoteAddr();
 
         clientIp = getOverrideHeader(request, overrideClientIpHeader, clientIp);
-        clientIp = getOverrideHeader(request, FORWARDED_HEADER, clientIp, false);
+        clientIp = getOverrideHeader(request, FORWARDED_FOR_HEADER, clientIp, false);
 
         return clientIp;
     }
 
+    /**
+     * Constructs the request URL based off of standard headers in the request, falling back to the HttpServletRequest.getRequestURL() 
+     * if the headers aren't available. Here is the ordered list of how we'll attempt to construct the URL: 
+     *  - x-api-request-url
+     *  - x-forwarded-proto://x-forwarded-host:x-forwarded-port/HttpServletRequest.getRequestURI()
```

Comment:
In my completely made-up requirements, yes.
Possible alternative would be to do something like:
 "if the host header is set and proto header isnt there and host header is not in requestUrl, then use the host from the host header and the protocol parse out of the requestUrl".

My inclination is that that is heavy handed and that users have to set x-forwarded-proto in their proxy, but I may be wrong here.